### ISSUE TITLE
would like support for "cmd <source>... <destination>"

### ISSUE THREAD

**Issue by tdterry** [docopt/docopt.go#5]
I have a cp-like command that takes multiple sources and a single destination. I'm trying to replicate that with docopt, but it consumes all of the input args for the <source>... match and then fails to match <destination>. The parser either needs a way to put back input tokens and try again (back tracking) or it needs to consider the next token (look ahead).

I tool a look at the source, for match(). I couldn't see any way to make this change. If you have any suggestions, I would be happy to work on a patch.

**Comment by badslug** [comment]
Rather than take into account all cases with a general solution, maybe the best place to start is to just handle the most common case: a pattern with a single `...` followed by a single non-optional last argument which is always parsed with the semantics of `cp`: one or more source items, and one final target item. It seems the vast majority of tools we would ever write falls into that one case. Maybe even more specifically we could make it parse only the option when that single ... is named "source" final non-optional argument is named "target". It's a special case (which is not ideal) but the special case covers 90% of actual use-cases.

**Comment by smaftoul** [comment]
any news on this ?
I also need a `source... target` positional argument.
Is this blocked by the python implementation to stay compatible ?

**Comment by smaftoul** [comment]
@mboersma any reasons you labeled this wontfix ?
Is it to keep this implementation compatible with python's one ?

**Comment by mboersma** [comment]
@smaftoul I think I marked this `wontfix` because of @keleshev's arguments above, and because yes, it would need to be supported in Python docopt first since all the implementations should be broadly the same.

Having said that, that decision was years ago, and in the meantime we've seen docopt stagnate. I think the implementations work reasonably well as-is, but there has been little maintenance. It seems unlikely we could get any changes made to the core Python implementation, and I'd rather not have the go implementation diverge from that. But I'm just someone who volunteered to help maintain this library and if there's a PR with community support I would certainly review it.
